MIPS/mip-zkapps.md

```diff
@@ -0,0 +1,961 @@
+owns the token to be minted; a child account update specifies that token and the
+number of tokens to be minted.
+
+This code, adapted from a Mina unit test, shows the pattern:
+
+```ocaml
+let token_minting =
+  mk_forest
+    [ mk_tree
+        (mk_account_update_body Signature No token_owner default_token
+           (- account_creation_fee) )
+        [ mk_tree
+           (mk_account_update_body None_given Parents_own_token
+             custom_token_account custom_token 100 )
+           []
+        ]
+    ]
+```
+
+`No` and `Parents_own_token` are the token permissions for the parent
+and child, respectively.  If these account updates succeed, a new
+account is created with a balance of 100 in the custom token.
+
+An account update using a custom token requires permission to use
+that token. In this example, the parent account is the the account id
+given by the public key `token_owner` and the token `default_token`;
+that account id owns `custom_token`. The permission
+`Parents_own_token` allows the new account to receive the minted
+`custom token`.
+
+### Permissions
+
+An account update can provide new values for certain account
+fields. Whether those changes are permitted is governed by a set of permissions.
+A permission can take on values `None`, `Either`, `Proof`, `Signature`, or
+`Impossible`, of OCaml type `Auth_required.t`.
+
+There is a set of associated tags `Proof`, `Signature`, and `None_given`. An
+account update that has a proof that verifies yields the tag `Proof`. A
+signature that verifies yields the tag `Signature`. If neither a proof nor a
+signature verifies, you get the `None_given` tag.
+
+Given a permission and a tag, an account update can perform an operation
+according to this OCaml function:
+
+```ocaml
+let check (t : Auth_required.t) (c : Control.Tag.t) =
+  match (t, c) with
+  | Impossible, _ ->
+      false
+  | None, _ ->
+      true
+  | Proof, Proof ->
+      true
+  | Signature, Signature ->
+      true
+  | Either, (Proof | Signature) ->
+      true
+  | Signature, Proof ->
+      false
+  | Proof, Signature ->
+      false
+  | (Proof | Signature | Either), None_given ->
+      false
+```
+
+The permissions that govern individual field updates are:
+
+- `edit_state`
+- `set_delegate`
+- `set_permissions`
+- `set_verification_key`
+- `set_zkapp_uri`
+- `edit_sequence_state`
+- `set_token_symbol`
+- `increment_nonce`
+- `set_voting_for`
+- `set_timing`
+
+There are permissions for sending and receiving tokens:
+
+- `send`
+- `receive`
+
+Finally, there is an `access` permission that allows an account
+update to access a ledger account. The `access` permission disallows all
+access to a ledger account if the authorization does not check against
+the permission, or the proof or signature authorization does not
+verify.
+
+The `access` permission introduced here affects not only zkApps, but
+also the payment and delegation transactions in the existing mainnet.
```

Comment:
```suggestion
also the payment and delegation transactions in the existing Mainnet.
```

Reply:
do we want to specify existing Mina Mainnet?

I capitalized Mainnet here to adhere to the guidance in the Mina docs style guide https://github.com/o1-labs/docs2/wiki/Docs-Style-Guide